test/hotspot/jtreg/compiler/c2/irTests/TestMinMaxIdentities.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.irTests;
+
+import jdk.test.lib.Asserts;
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8341781
+ * @summary Test identities of MinNodes and MaxNodes.
+ * @key randomness
+ * @requires (os.simpleArch == "x64" & vm.cpu.features ~= ".*avx.*") | os.arch == "aarch64" | os.arch == "riscv64"
```

Comment:
From my understanding this isn't possible as-is since CPU features seem to be checked regardless of whether the architecture supports it or not, so we can't simply check for AVX because that would fail on aarch64 and riscv64. I think we could work around this with `applyIfCPUFeatureOr = {"avx", "true", "asimd", "true", "rvv", "true"}` to force a check for all 3 platforms but it'd be filtering more platforms than strictly necessary.

Reply:
Which platforms would be filtered "more platforms than strictly necessary"?

Reply:
With the workaround to check for CPU features on all 3 platforms, we'd be not checking the IR when `asimd = false` or `rvv = false`, but the IR check should pass with those features too.

Reply:
I think your tags are actually available, with `applyIfPlatform`. Look at `irTestingPlatforms` in https://github.com/openjdk/jdk/blame/master/test/hotspot/jtreg/compiler/lib/ir_framework/test/IREncodingPrinter.java#L69

Reply:
Example:
`./test/hotspot/jtreg/compiler/loopopts/superword/TestGeneralizedReductions.java:    @IR(applyIfPlatform = {"riscv64", "true"},`

Reply:
The usage of these is quite rare - usually we focus more on the CPU features, and not the platform tags ;)

Reply:
Ah, I was meaning that with `applyIfPlatformOr = {"x64", "true", "aarch64", "true", "riscv64", "true"}` we would still need at least `applyIfCPUFeature = {"avx", true"}` because on x86 we only make Min/MaxF and Min/MaxD with AVX. But that applyIfCPUFeature will check for AVX on aarch64 and riscv64 as well, but will fail because it's not available for those platforms. That's why I suggested the workaround of checking the other CPU features, to make the test at least run on the other platforms. It'd be nice to be able to express platform and CPU feature combinations like with `@requires`, but the use-case here is pretty niche.

Reply:
First: yes, I'm fine with just using CPU features - it will IR test it on fewer platforms than maybe desired, but that is ok, I think.

I guess the issue with `@IR` is that it only allows `AND` or `OR` clauses ... and not the more complicated mix of `&` and `|` from `@requires`. But in theory, you can just have multiple `@IR` rules (that simulates the OR):
- One for `x64` (applyIfPlatform) and `avx` (applyIfCPUFeature)
- One IR rule for each of: `aarch64` and `riscv64` (applyIfPlatform)

But again, I'm ok with only checking for CPU features... it is more simple.

Reply:
@chhagedorn you may be interested in this conversation as well ;)

Reply:
I hadn't considered using multiple IR checks, that would definitely work as well. I've pushed a commit that just uses the CPU features, to keep it simple. Let me know what you think!